### PR TITLE
Initial Host connect

### LINKED ISSUE #172: Connect other host to view their posts

### LINKED ISSUE #173: Connect host to accpet and send friendrequest

### PR DESCRIPTION
**Pull request by zhaoyuxuan**
This PR includes:
1. Load posts from other host
2. Get authors from other hosts
3. Fixes the full id issue with user
4. Get friends of specific author
5. Rewrite Friendrequest API and tests
6. Fix author's url in the response

Test with 2 groups already.. works well
Here are the credientials from them

```
url: http://testing-mandala.herokuapp.com/posts/
username: admindemo
password: ualberta01!
(yes...thats the same as ours)
```

```
url:https://spongebook.herokuapp.com/
username(email): superman@email.com
password: passqwer

```
Insert those 2 into Host table to try it out

@roychowd @sandysaji  Please take a look on the implementation as you will be finishing the parts related to friends and authors. Let me know if there is any question or suggestion

### CODE REVIEW

File: backend/server.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    print(host.serviceAccountUsername)
+    response = requests.get(
+        host.url+endpoint,
+        auth=(host.serviceAccountUsername,
+              host.serviceAccountPassword)
+    )
+
+    return response
+
+# def get_host_data(request_endpoint):
```
Comment (roychowd) [inline backend/server.py]: can the commented block be deleted?

**Review by roychowd — APPROVED** [review]

**Review by roychowd — DISMISSED** [review]

**Review by sandysaji — DISMISSED** [review]

**Review by roychowd — DISMISSED** [review]
good work 👑

**Review by maharshmellow — DISMISSED** [review]
good shit

**Comment by ceciliaccwei** [comment]
Good shit

**Review by ceciliaccwei — COMMENTED** [review]
Good shit

**Review by maharshmellow — APPROVED** [review]

**Pull request merged**